### ISSUE TITLE
Fail to reproduce the results in paper

### ISSUE THREAD

**Issue by llCurious** [DachengLi1/MPCFormer#3]
Hey, @DachengLi1 . I try to reproduce the experiment results reported in the paper and found some mismatches.
![image](https://github.com/DachengLi1/MPCFormer/assets/21785695/16e947d1-4ebf-4ded-856a-a52778b9e40b)

1. MPCFormer of Bert-base on RTE task only achieves accuracy of 0.59, which should be 0.64. In comparison, Bert-base on STSB achieves 0.797, which is close to 0.803 in the paper. I use the defaut hyperparameter for RTE, do you have any ideas for this mismatch?
2. MPCFormer of Bert-large seems to diverge significantly from the paper. On RTE task, the acc drops to 0.47. I use quad+2quad approximation. I wonder if this is the root cause since the paper reported to use quad+2relu for Bert-large. Have you studied this point?

**Comment by DachengLi1** [comment]
@llCurious Nice to hear you are reproducing the results. I saw accuracy consistently robust for different hyper-parameters so this is not too expected.. Sometimes accuracy may seem to deviate by several points, and that is usually due to the nature of the dataset. For instance, CoLA and RTE can change by several points even you use the same training procedure but different seeds. MNLI, QQP, QNLI, SST2 are very robust. (and this is true not just for mpcformer, it is because the dataset).

(1) I get STS-B 79.9 for some hyper-parameters for STS-B, so this is good. It is matching. Specifically for RTE I don't have much ideas. I usually get 62-64, I never see 59.3. But I guess it doesn't matter much to just get RTE numbers exactly match since you have tested STS-B works? If you really want to debug into it, I guess (i) the teacher accuracy is different, I got a teacher with 69.7 accuracy. Since fine-tuning in RTE is fast, can you try to do the teacher training again and try to get this matching? Maybe it is simply because the teacher accuracy itself drops a bit already. (ii) try different random seeds for distillation.. 

(2) No, I simply didn't run quad+2quad for bert-large.. 0.47 means random guess in RTE. It means it completely breaks. Can you try (i) quad+2relu and see whether it is matching 72.2 (again I see the teacher model accuracy differs a bit, so I would expect something like 68-70.). (ii) quad+2quad for QNLI? this is a very simple dataset, if the first setting is working and QNLI for quad+2quad is not working even for QNLI, then there is some problem with the approximation. It may worth another paper to study when this framework can break.

Finally, if you are considering writing a new paper, let's sync up more (via email, zoom meeting, wechat etc) and I am willing to dig more into it.

**Comment by llCurious** [comment]
Thx to your quick reply. I tend to do some preliminary experiments on the approximation of non-linear activations in LLM. If it's convenient for you, may I add you on WeChat?  I email you my WeChat ID to the mail address listed in your Github page.

**Comment by DachengLi1** [comment]
Updates here - we are in contact for faster progress. Will update the issue once we solve it.

**Comment by DachengLi1** [comment]
Per @llCurious request, there might be an issue with current Bert-large implementation. Bert-Base is correct. I will reproduce myself and update later.

**Comment by DachengLi1** [comment]
Closing this as the author has already published a paper using MPCFormer as a baseline. There are no further questions, and seems this is due to random seed.
